### PR TITLE
Add reservable option to product, fixes #286

### LINKED ISSUE #286: Remove 'Reserve ticket' option
Currently, people are able to reserve tickets, but this causes some problems. For example, the board does not always know the exact amount of people that will attend. Money wise, it also gives some problems. We have so many ways you can now buy your tickets, which makes it very hard for the treasurer to keep track where the money came from.

I would suggest one of the following three options;
- Delete it (maybe for now only front end)
- Let the creator of the event decide whether they would like to have the reserved option.
- Make the reserved tickets expire 1 or 2 days in advance and send them a reminder to pay for their tickets.

It would be helpfull if one of those (or a better one) would be implemented.

### PR DESCRIPTION
**Pull request by Fastjur**
Create/update product view:
image

Payment view when an order contains a non-reservable product:
image

Signed-off-by: Jurriaan Den Toonder <jur.den.toonder@gmail.com>

### CODE REVIEW

File: src/main/resources/templates/webshop/payment/index.html [hunk 1]
```diff
@@ -114,6 +114,13 @@ <h5 class="d-inline mt-0">RESERVATION</h5> - Reserve your tickets now and
                                             </a>
                                         </div>
                                     </div>
+                                    <div class="row" th:if="${!containsOnlyReservable}">
```
Comment (svenpopping) [inline src/main/resources/templates/webshop/payment/index.html]: Think it is better to just remove the option, when it is not available.

File: src/main/resources/static/js/events/script.js [hunk 2]
```diff
@@ -43,6 +43,13 @@ var TemplateSelector;
                 } else {
                     inputBox.removeAttr("checked");
                 }
+
+                if (key === "reservable" && value === true) {
+                    inputBox.attr("checked", "checked");
+                    console.log('here');
```
Comment (svenpopping) [inline src/main/resources/static/js/events/script.js]: Remove logs

File: src/main/java/ch/wisv/events/core/service/order/OrderServiceImpl.java [hunk 3]
```diff
@@ -292,6 +292,16 @@ public boolean containsChOnlyProduct(Order order) {
         return order.getOrderProducts().stream().anyMatch(orderProduct -> orderProduct.getProduct().isChOnly());
     }
 
+    /**
+     * Returns whether an order contains non-reservable products
+     */
+    @Override
+    public boolean containsOnlyReservable(Order order) {
+        return order.getOrderProducts().stream().allMatch(orderProduct ->
```
[the remark this answers is not in the document]
Comment (Fastjur) [reply src/main/java/ch/wisv/events/core/service/order/OrderServiceImpl.java]: No it should not. This checks if there are reservable and reservable products in this order only.

**Comment by Fastjur** [comment]
The codeclimate issues don't make sense to me

File: src/main/java/ch/wisv/events/core/service/order/OrderServiceImpl.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (svenpopping) [reply src/main/java/ch/wisv/events/core/service/order/OrderServiceImpl.java]: Oh true true!

**Pull request merged**